ctapipe/image/cleaning.py

```diff
@@ -532,6 +596,49 @@ def __call__(
         )
 
 
+class TimeCleaner(ImageCleaner):
+    """
+    Clean images using the time clustering cleaning method
+    """
+
+    pedestal = FloatTelescopeParameter(
+        default_value=1.6, help="Pedestal width of each pixel"
+    ).tag(
+        config=True
+    )  # we should get this from a container once available
+    d_scale = FloatTelescopeParameter(
+        default_value=0.25, help="Pixel space scaling parameter in m"
+    ).tag(config=True)
+    t_scale = FloatTelescopeParameter(
+        default_value=4.0, help="Time scale parameter in ns"
+    ).tag(config=True)
+    minpts = IntTelescopeParameter(
+        default_value=5, help="minimum number of points to form a cluster"
+    ).tag(config=True)
+    eps = FloatTelescopeParameter(
+        default_value=1.0, help="minimum distance in DBSCAN"
+    ).tag(config=True)
+    noise_cut = FloatTelescopeParameter(
+        default_value=2.5, help="Number of sigmas for pre-cleaning the image"
+    ).tag(config=True)
+
+    def __call__(
+        self, tel_id: int, image: np.ndarray, arrival_times=None
+    ) -> np.ndarray:
+        """Apply HESS image cleaning. see ImageCleaner.__call__()"""
+        return time_clustering(
+            geom=self.subarray.tel[tel_id].camera.geometry,
+            image=image,
+            time=arrival_times,
+            eps=self.eps.tel[tel_id],
+            n_noise=self.noise_cut.tel[tel_id],
+            d_scale=self.d_scale.tel[tel_id],
+            t_scale=self.t_scale.tel[tel_id],
+            pedestal=self.pedestal.tel[tel_id],
```

Comment:
I think here you should really use the correct pedestal, not some hard-coded value. There is for example

```py3
ped = event.mon.tel[7].calibration.pedestal_per_sample
```
That is per channel, so you need to use the high/low gain switch mask.  We probably should properly process simulated pedestals into DL1 information though, so you don't have to do this.

Reply:
Maybe I should change the name but what I need is the reconstructed noise per pixel in units of pe. Is that available?

Reply:
Yeah, i realized that just after posting... the pedestals themselves are not what you need.  In the simulations currently we don't have the variances, but this is something we really need to fix.  Either we should properly compute pedestals and not rely on the ones given in the sims (which are technically inputs, not outputs), or see if somewhere the variances are already included or ask for them to be included.  That's not just or this PR, but for any "realistic" cleaning method, e.g. the Whipple-10m-style cleaning that is just tailcuts but with the thresholds expressed in pedestal variance units.

Reply:
You should definitely change the name, since "pedestal" is the mean value of the noise, not the dispersion.   Here, you are assuming it's Gaussian, right? And you are using it as a standard deviation, not variance, so maybe `pedestal_std`